r/vignettes/arrow.Rmd

```diff
@@ -216,7 +216,7 @@ Note that the `attributes()` stored in `$metadata$r` are only understood by R. I
 
 ## Class structure and package conventions
 
-C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented as `R6` reference classes, most of which are exported from the namespace.
+C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented using the `R6` object-oriented programming system, and most of them are exported from the namespace.
```

Comment:
```suggestion
C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented as `R6` classes, most of which are exported from the namespace.
```

Reply:
Could also link to the R6 package website

Reply:
Should we also link to Hadley's R6 (sub)chapter in _Advanced R_? Or would anyone interested find that as it's not really a hidden resource?